.github/actions/should_release/action.yml

```diff
@@ -0,0 +1,29 @@
+# TODO I feel like the wording on this is a little clunky and/or ethereal.
+# TODO we have is release vs should release and I don't feel like it's clear what the answer is
+name: Should release
+description: Determines if the current pull request is for testing or for starting a release
+outputs:
+    should_release:
+        description: Returns true if this branch should start a release, otherwise false
+        value: ${{ steps.determine_should_release.outputs.should_release }}
+runs:
+    using: composite
+    steps:
+        - uses: actions/checkout@v4
+          with:
+              ref: ${{ github.event.pull_request.head.ref }} # This is necessary for this job to be able to get the correct commit message from `git log`
+
+        - id: determine_should_release
+          run: |
+              BRANCH_NAME="${{ github.head_ref }}"
+              RELEASE_VERSION="${BRANCH_NAME:9}"
+              COMMIT_MESSAGE=$(git log -1 --pretty=format:"%s")
+              IS_RELEASE_BRANCH_PR="${{ contains(github.head_ref, 'release--') }}"
+
+              SHOULD_RELEASE="false"
+              if [[ $IS_RELEASE_BRANCH_PR == "true" && "$COMMIT_MESSAGE" != "azle-bot automated release $RELEASE_VERSION" ]]; then
```

Comment:
Create a declarative variable for the commit message text, the name should explain what that text is for

Reply:
Actually move this message up and pass it as an input to this action and the publish script. Call it: AUTOMATED_RELEASE_COMMIT_MESSAGE

Reply:
Actually, don't do any of that. Change the if statement to check for an exact message, which will be the name of the release branch (e.g. release--0.25.0-rc.0)

Reply:
done